File: ios/Classes/streaming/utils/ParsedDimensions.m
#import "ParsedDimensions.h"

@implementation ParsedDimensions;


- (ParsedDimensions*)initWithData:(id) data {
    self = [super init];
    if(self) {
        if (data == [NSNull null] || data == nil) {
            self.width = -1;
            self.height = -1;
            self.isValid = NO;
        } else {
            self.width = data[@"width"] != [NSNull null] ? [data[@"width"] intValue] : -1;
            self.height = data[@"height"] != [NSNull null] ? [data[@"height"] intValue] : -1;
            self.isValid = !(self.width < 0 || self.height < 0);
        }
    }
    return self;
}

@end

File: android/src/main/java/com/comscore/flutter/streaming/utils/ParsedDimensions.java
package com.comscore.flutter.streaming.utils;

import static com.comscore.flutter.Args.getArgValue;

public class ParsedDimensions {
    public final Integer width, height;
    public ParsedDimensions(Object data) {
        if (data == null) {
            width = null;
            height = null;
            return;
        }
        width = getArgValue(data, "width");
        height = getArgValue(data, "height");
    }

    public boolean isValid() {
        return width != null && height != null;
    }
}
